File: python-sdk/src/boxsdk/models/operations/post_retention_policies.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import user_mini as shared_user_mini
from ..shared import clienterror as shared_clienterror
from ..shared import retentionpolicy as shared_retentionpolicy

class PostRetentionPoliciesRequestBodyDispositionActionEnum(str, Enum):
    PERMANENTLY_DELETE = "permanently_delete"
    REMOVE_RETENTION = "remove_retention"

class PostRetentionPoliciesRequestBodyPolicyTypeEnum(str, Enum):
    FINITE = "finite"
    INDEFINITE = "indefinite"

class PostRetentionPoliciesRequestBodyRetentionTypeEnum(str, Enum):
    MODIFIABLE = "modifiable"
    NON_MODIFIABLE = "non-modifiable"


@dataclass_json
@dataclasses.dataclass
class PostRetentionPoliciesRequestBody:
    disposition_action: PostRetentionPoliciesRequestBodyDispositionActionEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('disposition_action') }})
    policy_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('policy_name') }})
    policy_type: PostRetentionPoliciesRequestBodyPolicyTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('policy_type') }})
    are_owners_notified: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('are_owners_notified') }})
    can_owner_extend_retention: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('can_owner_extend_retention') }})
    custom_notification_recipients: Optional[list[shared_user_mini.UserMini]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('custom_notification_recipients') }})
    description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    retention_length: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('retention_length') }})
    retention_type: Optional[PostRetentionPoliciesRequestBodyRetentionTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('retention_type') }})
    

@dataclasses.dataclass
class PostRetentionPoliciesRequest:
    request: Optional[PostRetentionPoliciesRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PostRetentionPoliciesResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    retention_policy: Optional[shared_retentionpolicy.RetentionPolicy] = dataclasses.field(default=None)
    

File: ts-sdk/src/sdk/models/operations/postretentionpolicies.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum PostRetentionPoliciesRequestBodyDispositionActionEnum {
    PermanentlyDelete = "permanently_delete",
    RemoveRetention = "remove_retention"
}

export enum PostRetentionPoliciesRequestBodyPolicyTypeEnum {
    Finite = "finite",
    Indefinite = "indefinite"
}

export enum PostRetentionPoliciesRequestBodyRetentionTypeEnum {
    Modifiable = "modifiable",
    NonModifiable = "non-modifiable"
}


export class PostRetentionPoliciesRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=are_owners_notified" })
  areOwnersNotified?: boolean;

  @SpeakeasyMetadata({ data: "json, name=can_owner_extend_retention" })
  canOwnerExtendRetention?: boolean;

  @SpeakeasyMetadata({ data: "json, name=custom_notification_recipients", elemType: shared.UserMini })
  customNotificationRecipients?: shared.UserMini[];

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=disposition_action" })
  dispositionAction: PostRetentionPoliciesRequestBodyDispositionActionEnum;

  @SpeakeasyMetadata({ data: "json, name=policy_name" })
  policyName: string;

  @SpeakeasyMetadata({ data: "json, name=policy_type" })
  policyType: PostRetentionPoliciesRequestBodyPolicyTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=retention_length" })
  retentionLength?: string;

  @SpeakeasyMetadata({ data: "json, name=retention_type" })
  retentionType?: PostRetentionPoliciesRequestBodyRetentionTypeEnum;
}


export class PostRetentionPoliciesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostRetentionPoliciesRequestBody;
}


export class PostRetentionPoliciesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  retentionPolicy?: shared.RetentionPolicy;

  @SpeakeasyMetadata()
  statusCode: number;
}
